### ISSUE TITLE
Heroku Deployment

### ISSUE THREAD

**Issue by AdamSaketume26** [jef/nvidia-snatcher#497]
Did anyone manage to setup nvidia snatcher on Heroku? I am having issues with getting it to work because of typescript. It seems like everything I try doesn't help either.

**Comment by gperl27** [comment]
Was able to get it working on Heroku by doing the following:

1. Create a `Procfile` in the root directory of the project and add the following line:

`worker: npm run start:production`

We use `worker` or else Heroku will timeout by attempting to bind to a `port`. Also notice we're using the `start:production` target. When you push to Heroku, your code is already compiled by `tsc`.

2. Update `puppeteer` args in the `main` function in `index.ts`
```
const args: string[] = [
  '--no-sandbox',
  '--disable-setuid-sandbox'
];
```

3. Add the puppeteer buildpack in settings in your Heroku dashboard:

Manually add `https://github.com/jontewks/puppeteer-heroku-buildpack` to the "buildpacks" list

**Comment by jef** [comment]
This is good! Maybe we should add into the wiki. Thank you!

**Comment by gperl27** [comment]
Awesome, looks good. Glad I could help!

**Comment by GamerKingFaiz** [comment]
Thanks for the additional instructions @gperl27!

I got the app to run on Heroku, but I'm getting the following error when there's something in stock and the app tries to send a notification (I only have Discord set up).
```
[11:57:14 PM] [32minfo[39m :: 🚀🚨 [walmart] [microsoft (xboxss)] xbox series s :: IN STOCK 🚨🚀
https://www.walmart.com/ip/Xbox-Series-S/606518560
node:events:342
      throw er; // Unhandled 'error' event
      ^

Error: spawn xdg-open ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn xdg-open',
  path: 'xdg-open',
  spawnargs: [ 'https://www.walmart.com/ip/Xbox-Series-S/606518560' ]
}
```

I also get a similar error when I run the test notification command (`npm run test:notification:production`).
```
node:events:342
      throw er; // Unhandled 'error' event
      ^

Error: spawn xdg-open ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:282:19)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:12)
    at onErrorNT (node:internal/child_process:480:16)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn xdg-open',
  path: 'xdg-open',
  spawnargs: [ 'https://www.example.com/cartUrl' ]
}
```

For background, I followed @gperl27's steps.
I also needed to add the `heroku/nodejs` buildpack in my Heroku app's settings, otherwise it wasn't finding the node command.
Lastly, I changed my `.gitignore` to upload the `build/` folder and the `dotenv` file because the app needs these in Heroku.

The app works completely fine locally on my Windows machine, but I'm not able to get it to run with the same configuration in Heroku. Is there anything I'm missing here?

**Comment by GamerKingFaiz** [comment]
Fixed the issue thanks to #2269!

> had to add OPEN_BROWSER=false to the env file.
